adapters/beachfront/beachfront.go

```diff
@@ -291,195 +285,215 @@ func getBannerRequest(request *openrtb.BidRequest) (beachfrontBannerRequest, []e
-			bfr.User.ID = request.User.ID
+		if bannerReq.User.ID == "" {
+			bannerReq.User.ID = request.User.ID
 		}
 	}
 
 	if request.User != nil && request.User.BuyerUID != "" {
-		if bfr.User.BuyerUID == "" {
-			bfr.User.BuyerUID = request.User.BuyerUID
+		if bannerReq.User.BuyerUID == "" {
+			bannerReq.User.BuyerUID = request.User.BuyerUID
 		}
 	}
 
-	bfr.RequestID = request.ID
-	bfr.AdapterName = beachfrontAdapterName
-	bfr.AdapterVersion = beachfrontAdapterVersion
-
-	if request.Imp[0].Secure != nil {
-		bfr.Secure = *request.Imp[0].Secure
-	}
+	bannerReq.RequestID = request.ID
+	bannerReq.AdapterName = beachfrontAdapterName
+	bannerReq.AdapterVersion = beachfrontAdapterVersion
 
-	return bfr, errs
+	return
 }
 
-func fallBackDeviceType(request *openrtb.BidRequest) openrtb.DeviceType {
-	if request.Site != nil {
-		return openrtb.DeviceTypePersonalComputer
-	}
+func getVideoRequests(requestStub openrtb.BidRequest, imps []openrtb.Imp) (nurlReqs []videoRequest, admReqs []videoRequest, errs []error) {
+	var ext openrtb_ext.ExtImpBeachfront
+	var admMap = make(map[string]videoRequest)
 
-	return openrtb.DeviceTypeMobileTablet
-}
+	for i := 0; i < len(imps); i++ {
+		ext, errs = prepVideoRequestExt(imps[i], errs)
 
-func getVideoRequests(request *openrtb.BidRequest) ([]beachfrontVideoRequest, []error) {
-	var bfReqs = make([]beachfrontVideoRequest, len(request.Imp))
-	var errs = make([]error, 0, len(request.Imp))
-	var failedRequestIndicies = make([]int, 0)
+		if ext.VideoResponseType == "nurl" || ext.VideoResponseType == "both" {
+			requestStub.Imp = []openrtb.Imp{imps[i]}
 
-	for i := 0; i < len(request.Imp); i++ {
-		beachfrontExt, err := getBeachfrontExtension(request.Imp[i])
+			r := videoRequest{
+				AppId:             ext.AppId,
+				VideoResponseType: ext.VideoResponseType,
+				Request:           requestStub,
+			}
 
-		if err != nil {
-			// Failed to extract the beachfrontExt, so this request is junk.
-			failedRequestIndicies = append(failedRequestIndicies, i)
-			errs = append(errs, err)
-			continue
+			prepVideoRequest(&r)
+			nurlReqs = append(nurlReqs, r)
 		}
 
-		appid, err := getAppId(beachfrontExt, openrtb_ext.BidTypeVideo)
+		if ext.VideoResponseType == "adm" || ext.VideoResponseType == "both" {
+			_, exists := admMap[ext.AppId]
+			if !exists {
+				requestStub.Imp = make([]openrtb.Imp, 0, len(imps))
+				requestStub.Imp = append(requestStub.Imp, imps[i])
 
-		if err != nil {
-			// Failed to get an appid, so this request is junk.
-			failedRequestIndicies = append(failedRequestIndicies, i)
-			errs = append(errs, err)
-			continue
+				admRequest := videoRequest{
+					AppId:             ext.AppId,
+					VideoResponseType: ext.VideoResponseType,
+					Request:           requestStub,
+				}
+				prepVideoRequest(&admRequest)
+				admMap[ext.AppId] = admRequest
+			} else {
+				tmpRequest := admMap[ext.AppId].Request
+				tmpRequest.Imp = append(tmpRequest.Imp, imps[i])
+
+				admMap[ext.AppId] = videoRequest{
```

Comment:
Any reason why you need to reassign the request in `admMap`? You have already appended the new imp in above two lines. All the other values for the request should already be same, no?